File: app/src/main/java/org/stepic/droid/util/DbParseHelper.java
package org.stepic.droid.util;

public class DbParseHelper {

    private static final String DELIMITER = "__,__";

    public static long[] parseStringToLongArray(String str) {
        if (str == null) return null;

        String[] strArray = str.split(DELIMITER);
        long[] result = new long[strArray.length];
        for (int i = 0; i < strArray.length; i++)
            result[i] = Long.parseLong(strArray[i].trim());
        return result;
    }

    public static String parseLongArrayToString(long[] array) {
        return parseLongArrayToString(array, DELIMITER);
    }

    public static String parseLongArrayToString(long[] array, String delimiter) {
        if (array == null || array.length == 0) return null;

        StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < array.length; i++) {
            stringBuilder.append(array[i]);
            if (i != array.length - 1)
                stringBuilder.append(delimiter);
        }
        return stringBuilder.toString();
    }


    public static String parseStringArrayToString(String[] array) {
        if (array == null || array.length == 0) return null;

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < array.length; i++) {
            sb.append(array[i]);
            if (i != array.length - 1)
                sb.append(DELIMITER);
        }
        return sb.toString();
    }

    public static String[] parseStringToStringArray(String str) {
        if (str == null) return null;

        String[] strArray = str.split(DELIMITER);
        String[] result = new String[strArray.length];
        for (int i = 0; i < strArray.length; i++)
            result[i] = strArray[i].trim();
        return result;
    }

    public static String[] parseLongArrayToStringArray(long[] array) {
        String[] result = new String[array.length];
        for (int i = 0; i < array.length; i++) {
            result[i] = array[i] + "";
        }
        return result;
    }

    public static String parseLongArrayToString(Long[] array, String delimiter) {

        if (array == null || array.length == 0) return null;

        StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < array.length; i++) {
            stringBuilder.append(array[i]);
            if (i != array.length - 1)
                stringBuilder.append(delimiter);
        }
        return stringBuilder.toString();
    }


}

File: app/src/main/java/org/stepic/droid/util/DbParseHelper.kt
package org.stepic.droid.util

object DbParseHelper {

    private const val DEFAULT_SEPARATOR = "__,__"
    private const val PREFIX = "["
    private const val POSTFIX = "]"

    @JvmStatic
    @JvmOverloads
    fun parseStringToLongArray(str: String?, separator: String = DEFAULT_SEPARATOR, escapeSymbols: Boolean = false): LongArray? =
            parseStringToLongList(str, separator, escapeSymbols)?.toLongArray()

    @JvmStatic
    @JvmOverloads
    fun parseStringToLongList(str: String?, separator: String = DEFAULT_SEPARATOR, escapeSymbols: Boolean = false): List<Long>? =
            try {
                str?.split(separator)?.map {
                    if (escapeSymbols) {
                        it.trim().removeSurrounding(PREFIX, POSTFIX).toLong()
                    } else {
                        it.trim().toLong()
                    }
                }
            } catch (e: Exception) {
                null
            }

    @JvmStatic
    @JvmOverloads
    fun parseLongArrayToString(longArray: LongArray?, separator: String = DEFAULT_SEPARATOR, escapeSymbols: Boolean = false): String? {
        if (longArray == null || longArray.isEmpty()) return null

        return if (escapeSymbols) {
            longArray.joinToString(separator, transform = { "$PREFIX$it$POSTFIX" } )
        } else {
            longArray.joinToString(separator)
        }
    }

    @JvmStatic
    @JvmOverloads
    fun parseLongListToString(longArray: List<Long>?, separator: String = DEFAULT_SEPARATOR, escapeSymbols: Boolean = false): String? {
        if (longArray == null || longArray.isEmpty()) return null

        return if (escapeSymbols) {
            longArray.joinToString(separator, transform = { "$PREFIX$it$POSTFIX" })
        } else {
            longArray.joinToString(separator)
        }
    }

    @JvmStatic
    @JvmOverloads
    fun parseStringArrayToString(array: Array<String>?, separator: String = DEFAULT_SEPARATOR): String? {
        if (array == null || array.isEmpty()) return null

        return array.joinToString(separator)
    }

    @JvmStatic
    @JvmOverloads
    fun parseStringToStringList(str: String?, separator: String = DEFAULT_SEPARATOR): List<String>? =
            str?.split(separator)?.map(String::trim)


    @JvmStatic
    @JvmOverloads
    fun parseStringToStringArray(str: String?, separator: String = DEFAULT_SEPARATOR): Array<String>? =
            parseStringToStringList(str, separator)?.toTypedArray()

}


